src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,157 @@
 package core.basesyntax;
 
+import java.util.Map;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    static final int DEFAULT_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private float loadFactor;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        this.loadFactor = DEFAULT_LOAD_FACTOR;
+        this.threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
 
+        Node<K, V> current = table[index];
+        while (true) {
+            if ((current.getKey() == key) || (key != null && key.equals(current.getKey()))) {
```

Comment:
```suggestion
            if (current.getKey() == key || key != null && key.equals(current.getKey())) {
```
Redundant brackets. read about logical operator priority